zstash/hpss.py

```diff
@@ -113,10 +119,15 @@ def hpss_transfer(
         if transfer_type == "put":
             if not keep:
                 if (scheme != "globus") or (
-                    globus_status == "SUCCEEDED" and not non_blocking
+                    globus_status == "SUCCEEDED"
                 ):
-                    os.remove(file_path)
-
+                    # Note: This is intended to fulfill the default removal of successfully-transfered
+                    # tar files when keep=False, irrespective of non-blocking status
+                    logger.info(f"{ts_utc()}: DEBUG: deleting transfered files {prev_transfers}")
+                    for src_path in prev_transfers:
+                        os.remove(src_path)
+                    prev_transfers = curr_transfers
+                    curr_transfers = list()
```

Comment:
You can just use `= []` instead of `= list()`.

Reply:
I used to do that - but was cautioned against it (don't recall why).  I'd be happy either way.

Reply:
Hmm interesting, I wonder why. `= []` definitely seems more "pythonic" to me, as is echoed on https://stackoverflow.com/questions/5790860/whats-the-difference-between-and-vs-list-and-dict.